tests/unit/test-cmd/test.run.js

```diff
@@ -1,35 +1,43 @@
 /* @flow */
 import path from 'path';
-import EventEmitter from 'events';
-import tty from 'tty';
 import stream from 'stream';
+import tty from 'tty';
 
 import {describe, it} from 'mocha';
 import {assert} from 'chai';
 import sinon from 'sinon';
 
-import {onlyInstancesOf, WebExtError, RemoteTempInstallNotSupported}
-  from '../../../src/errors';
 import run, {
-  defaultFirefoxClient, defaultWatcherCreator, defaultReloadStrategy,
+  defaultWatcherCreator,
+  defaultReloadStrategy,
   defaultAddonReload,
 } from '../../../src/cmd/run';
-import * as defaultFirefoxApp from '../../../src/firefox';
-import {RemoteFirefox} from '../../../src/firefox/remote';
-import {TCPConnectError, fakeFirefoxClient, makeSureItFails, fake, fixturePath}
-  from '../helpers';
+import {
+  fixturePath,
+  makeSureItFails,
+  FakeExtensionRunner,
+  getFakeFirefox,
+  getFakeRemoteFirefox,
+} from '../helpers';
 import {createLogger} from '../../../src/util/logger';
 
 const log = createLogger(__filename);
 // Fake result for client.installTemporaryAddon().then(installResult => ...)
 const tempInstallResult = {
   addon: {id: 'some-addon@test-suite'},
 };
-// Fake missing addon id result for client.installTemporaryAddon
-const tempInstallResultMissingAddonId = {
-  addon: {id: null},
-};
 
+function createFakeExtensionRunner(params, deps, overriddenMethods = {}) {
```

Comment:
It looks like `params` and `deps` are merged together in the real implementation now.